applications/StructuralMechanicsApplication/custom_constitutive/linear_isotropic_damage_3D_law.cpp

```diff
@@ -0,0 +1,401 @@
+    const Vector& rShapeFunctionsValues
+    )
+{
+    const double yield_stress = rMaterialProperties[YIELD_STRESS];
+    const double young_modulus = rMaterialProperties[YOUNG_MODULUS];
+    r_prev = yield_stress / std::sqrt(young_modulus);
+}
+
+//************************************************************************************
+//************************************************************************************
+
+void LinearIsotropicDamage3D::FinalizeSolutionStep(
+    const Properties& rMaterialProperties,
+    const GeometryType& rElementGeometry,
+    const Vector& rShapeFunctionsValues,
+    const ProcessInfo& rCurrentProcessInfo)
+{
+    // update of damage threshold
+    r_prev = r;
+}
+
+//************************************************************************************
+//************************************************************************************
+
+void LinearIsotropicDamage3D::CalculateMaterialResponsePK1(Parameters& rValues)
+{
+    CalculateMaterialResponseCauchy(rValues);
+}
+
+//************************************************************************************
+//************************************************************************************
+
+void LinearIsotropicDamage3D::CalculateMaterialResponsePK2(Parameters& rValues)
+{
+    CalculateMaterialResponseCauchy(rValues);
+}
+
+//************************************************************************************
+//************************************************************************************
+
+void LinearIsotropicDamage3D::CalculateMaterialResponseKirchhoff(Parameters& rValues)
+{
+    CalculateMaterialResponseCauchy(rValues);
+}
+
+//************************************************************************************
+//************************************************************************************
+
+void LinearIsotropicDamage3D::CalculateMaterialResponseCauchy(Parameters& rValues)
+{
+    const Flags &Options = rValues.GetOptions();
+
+    const Properties& rMaterialProperties = rValues.GetMaterialProperties();
+    Vector& strain_vector = rValues.GetStrainVector();
+    Vector& stress_vector = rValues.GetStressVector();
+
+    if (Options.Is(ConstitutiveLaw::COMPUTE_CONSTITUTIVE_TENSOR)) {
+        Matrix& constitutive_matrix = rValues.GetConstitutiveMatrix();
+        CalculateConstitutiveMatrix(constitutive_matrix, rMaterialProperties);
+    }
+
+    if (Options.Is(ConstitutiveLaw::COMPUTE_STRESS)) {
+        if (rValues.GetProcessInfo().Has(INITIAL_STRAIN)) {
+            noalias(strain_vector) += rValues.GetProcessInfo()[INITIAL_STRAIN];
+        }
+        Matrix& constitutive_matrix = rValues.GetConstitutiveMatrix();
+
+        CalculateConstitutiveMatrix(constitutive_matrix, rMaterialProperties);
+        noalias(stress_vector) = prod(constitutive_matrix, strain_vector);
+        // For use in TRACTION_ONLY case.
+        Vector stress_vector_pos = prod(constitutive_matrix, strain_vector);
+
```

Comment:
It is elastic behaviour?
Maybe you can used the function of the elastic CL

Reply:
Do you really need this vector `stress_vector_pos`?

Reply:
stress_vector_pos is not needed now, will be used in the future. Removing it.